homeassistant/components/freebox/base_class.py

```diff
@@ -0,0 +1,197 @@
+"""Support for Freebox base features."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, VALUE_NOT_SET
+from .router import FreeboxRouter
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FreeboxHomeBaseClass(Entity):
+    """Representation of a Freebox base entity."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant | None,
+        router: FreeboxRouter,
+        node: dict[str, Any],
+        sub_node=None,
+    ) -> None:
+        """Initialize a Freebox entity."""
+        self._hass = hass
+        self._router = router
+        self._node = node
+        self._sub_node = sub_node
+        self._id = node["id"]
+        self._name = node["label"].strip()
+        self._device_name = node["label"].strip()
+        self._unique_id = f"{self._router.mac}-node_{self._id}"
+
+        if sub_node is not None:
+            self._name = node["label"].strip() + " " + sub_node["label"].strip()
+            self._unique_id += "-" + sub_node["name"].strip()
+
+        self._available = True
+        self._firmware = node["props"].get("FwVersion")
+        self._manufacturer = "Freebox SAS"
+        self._model = ""
+        self._remove_signal_update: Any
+
+        if node["category"] == "pir":
+            self._model = "F-HAPIR01A"
+        elif node["category"] == "camera":
+            self._model = "F-HACAM01A"
+        elif node["category"] == "dws":
+            self._model = "F-HADWS01A"
+        elif node["category"] == "kfb":
+            self._model = "F-HAKFB01A"
+        elif node["category"] == "alarm":
```

Comment:
Why not using switch statement ?

Reply:
hi!
Thanks for you feed back.... hum! good question.... 
For next update, I replaced the if/else block code by

```suggestion
        match node["category"]:
            case "pir":
                self._model = "F-HAPIR01A"
            case "camera":
                self._model = "F-HACAM01A"
            case "dws":
                self._model = "F-HADWS01A"
            case "kfb":
                self._model = "F-HAKFB01A"
            case "alarm":
                self._model = "F-MSEC07A"
            case _:
                if node["type"].get("inherit") == "node::rts":
                    self._manufacturer = "Somfy"
                    self._model = "RTS"
                elif node["type"].get("inherit") == "node::ios":
                    self._manufacturer = "Somfy"
                    self._model = "IOHome"
```

Reply:
The same in homeassistant/components/freebox/camera.py file for this if/elif block
```
           if endpoint["name"] == ATTR_DETECTION:
                self._attr_motion_detection_enabled = endpoint["value"]
            elif endpoint["name"] == ATTR_ACTIVATION:
                self._activation_with_alarm = endpoint["value"]
            elif endpoint["name"] == ATTR_QUALITY:
                self._high_quality_video = endpoint["value"]
            elif endpoint["name"] == ATTR_SENSITIVITY:
                self._motion_sensitivity = endpoint["value"]
            elif endpoint["name"] == ATTR_THRESHOLD:
                self._motion_threshold = endpoint["value"]
            elif endpoint["name"] == ATTR_FLIP:
                self._flip = endpoint["value"]
            elif endpoint["name"] == ATTR_TIMESTAMP:
                self._timestamp = endpoint["value"]
            elif endpoint["name"] == ATTR_VOLUME:
                self._volume_micro = endpoint["value"]
            elif endpoint["name"] == ATTR_SOUND_DETECTION:
                self._sound_detection = endpoint["value"]
            elif endpoint["name"] == ATTR_SOUND_TRIGGER:
                self._sound_trigger = endpoint["value"]
            elif endpoint["name"] == ATTR_RTSP:
                self._rtsp = endpoint["value"]
            elif endpoint["name"] == ATTR_DISK:
                self._disk = endpoint["value"]
```

Reply:
The Python idiom for this is a dict lookup, not if-else or pattern matching:

```py
CATEGORY_TO_MODEL = {
    "pir": "F-HAPIR01A",
    ...
}
```

Reply:
Thanks..


```
CATEGORY_TO_MODEL = {
    "pir": "F-HAPIR01A",
    "camera": "F-HACAM01A",
    "dws": "F-HADWS01A",
    "kfb": "F-HAKFB01A",
    "alarm": "F-MSEC07A",
}
```


```
        self._model = CATEGORY_TO_MODEL.get(node["category"])
        if self._model is None:
            if node["type"].get("inherit") == "node::rts":
                self._manufacturer = "Somfy"
                self._model = "RTS"
            elif node["type"].get("inherit") == "node::ios":
                self._manufacturer = "Somfy"
                self._model = "IOHome"
```

Reply:
Comment not resolved or tell why it should be

Reply:
Comment not resolved: PLEASE DON'T RESOLVE IF NOT RESOLED !

Better: let reviewer resolve his comments !

Reply:
Fixed